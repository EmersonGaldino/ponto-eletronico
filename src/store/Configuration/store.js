import { createStore, applyMiddleware, compose } from 'redux';
import combinedReducers from './combinedReducers';
import rootSaga from './rootSaga';
import createSaga from 'redux-saga';
import { loadState, saveState } from '../../common/Utils';
import throttle from 'lodash.throttle';

const sagaMiddleware = createSaga();

const persistedState = loadState();

const store = createStore(
  combinedReducers,
  persistedState,
  compose(applyMiddleware(sagaMiddleware))
);

store.subscribe(
  throttle(() => {
    saveState({
      ...store.getState(),
    });
  }, 1000)
);

sagaMiddleware.run(rootSaga);

export { store };
