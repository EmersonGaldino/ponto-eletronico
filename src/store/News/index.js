import {NewsTypes} from './types';

const INITIAL_STATE = {
    point: [],
    loading: false,
    error: false,
    success: false,
    status:'',
    totalResults:0,
    articles: []
};

const reducer = (state = INITIAL_STATE, action) => {
    switch (action.type) {
        case NewsTypes.NEWS_REQUEST:
            return {...state, loading: true};

        case NewsTypes.NEWS_LOAD_SUCCESS:
            return {
                ...state,
                loading: false,
                success: true,
                status:action.payload.data.status,
                totalResults:action.payload.data.totalResults,
                articles: action.payload.data.articles
            };

        case NewsTypes.NEWS_LOAD_FAILURE:
            return {...state, loading: false, error: true, success: false};

        default:
            return state;
    }
};

export default reducer;
