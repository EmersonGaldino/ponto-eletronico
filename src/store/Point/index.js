import {PointTypes} from './types';

const INITIAL_STATE = {
    point: [],
    loading: false,
    error: false,
    success: false,
    hours:0,
    points: []
};

const reducer = (state = INITIAL_STATE, action) => {
    switch (action.type) {
        case PointTypes.POINT_REQUEST:
            return {...state, loading: true};

        case PointTypes.POINT_LOAD_SUCCESS:
            return {
                ...state,
                loading: false,
                success: true,
                points: action.payload.data,
            };

        case PointTypes.POINT_REGISTER:
            return {...state, loading: true};

        case PointTypes.POINT_REGISTER_SUCCESS:
            return {
                ...state,
                loading: false,
                success: true,
                point: action.payload.data,
            };

        case PointTypes.POINT_LOAD_FAILURE:
            return {...state, loading: false, error: true, success: false};

        case PointTypes.POINT_REGISTER_FAILURE:
            return {...state, loading: false, error: true, success: false};

        default:
            return state;
    }
};

export default reducer;
