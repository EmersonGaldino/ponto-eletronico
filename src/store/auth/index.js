import { AuthTypes } from './types';

const INITIAL_STATE = {
  loading: false,
  success: false,
  accessToken: '',
  refreshToken: '',
  userInfo: {},
  profile:{},
  message:''
};

const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AuthTypes.AUTH_REFRESH:
      return {
        ...state,
        token: action.payload.token,
      };
    case AuthTypes.AUTH_STATE:
      return {
        ...state,
        [action.payload.node]: {
          ...state[action.payload.node],
          [action.payload.key]: action.payload.value,
        },
      };
    case AuthTypes.SIGN_IN_START:
      return {
        ...state,
        loading: true,
        success: false,
        accessToken: undefined,
        refreshToken: undefined,
        userInfo: {},
        profile: {}
      };
    case AuthTypes.SIGN_IN_REQUEST:
      return {
        ...state,
        loading: true,
        success: false,
        accessToken: undefined,
        refreshToken: undefined,
        message:'',
        userInfo: {},
        profile: {}
      };
    case AuthTypes.SIGN_IN_SUCCESS:
      return {
        ...state,
        userInfo: action.payload.userInfo,
        profile: action.payload.profile,
        accessToken: action.payload.accessToken,
        refreshToken: action.payload.refreshToken,
        loading: false,
        success: true,
        message:''
      };
    case AuthTypes.SIGN_IN_FAILURE:
      return {
        ...state,
        loading: false,
        success: false,
        accessToken: undefined,
        userInfo: {},
        profile: {},
        message: 'Login n√£o autorizado.'
      };
    case AuthTypes.SIGN_OUT: {
      return { ...state };
    }
    case AuthTypes.FIRST_ACCESS:
      return { ...state, userInfo: { ...state.userInfo, firstAccess: false } };
    default:
      return state;
  }
};

export default reducer;
