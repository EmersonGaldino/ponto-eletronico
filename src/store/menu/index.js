import { MenuTypes } from './types';

const INITIAL_STATE = {
  menu: [],
  loading: false,
  error: false,
  success:false,
  selected:'Home',
};

const reducer = (state = INITIAL_STATE, action) => {
  console.log(action)
  switch (action.type) {
    case MenuTypes.MENU_REQUEST:
      return { ...state, loading: true };
    case MenuTypes.MENU_LOAD_SUCCESS:
      return {
        ...state,
        loading: false,
        success: true,
        selected:'',
        menu: action.payload.data.objectReturn,
      };
    case MenuTypes.MENU_LOAD_FAILURE:
      return { ...state, loading: false, error: true, success: false,menu:[] };
    default:
      return state;
  }
};

export default reducer;
