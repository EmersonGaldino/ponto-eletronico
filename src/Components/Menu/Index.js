import React, {useState, useEffect} from 'react';
import {Menu, Segment, Dropdown} from 'semantic-ui-react';
import {connect} from 'react-redux';
import * as MenuActions from '../../store/menu/actions';
import * as AuthActions from '../../store/auth/actions';
import {Link, useHistory} from 'react-router-dom';

const Index = ({menu, userInfo, logOut, success}) => {
    const history = useHistory();
    const [active, setActive] = useState('Home');
    const [menuItem, setMenu] = useState([]);

    useEffect(() => {
        setMenu(menu);
    }, [success]);

    const LogOutAsync = () => {
        logOut();
        history.push('/');
    }

    const redirectPage = (item) => {
        setActive(item.description);
        history.push(item.link)
    }

    return (
        <Menu inverted fixed={"top"}>
            {menuItem.map((item) =>
                item.childrens != 0 ? (
                    <Dropdown item text={item.description} key={item.id}>
                        <Dropdown.Menu>
                            <Dropdown.Header name={item.description}>
                                {item.description}
                            </Dropdown.Header>
                            {item.childrens.map((subMenu, index) => (
                                <Dropdown.Item key={index}
                                               onClick={() => <Link to={item.link}/>} name={item.description}>
                                    {subMenu.description}
                                </Dropdown.Item>
                            ))}
                        </Dropdown.Menu>
                    </Dropdown>
                ) : (
                    <Menu.Item
                        name={item.description}
                        active={active === item.description}
                        onClick={() => redirectPage(item)}
                    />
                )
            )}
            <Menu.Menu position='right'>
                <Dropdown item text={userInfo.name}>
                    <Dropdown.Menu>
                        <Dropdown.Header>Actions</Dropdown.Header>
                        <Link to='/profile' >
                            <Dropdown.Item style={{color:'#000'}}>Perfil</Dropdown.Item>
                        </Link>
                        <Dropdown.Item onClick={() => LogOutAsync()}>Sair</Dropdown.Item>
                    </Dropdown.Menu>
                </Dropdown>
            </Menu.Menu>
        </Menu>
    );
};
const mapStateToProps = (state) => {
    return {
        userInfo: state.auth.userInfo,
        token: state.auth.accessToken,
        menu: state.menu.menu,
        success: state.menu.success,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        loadingMenu: (token) => {
            dispatch(MenuActions.loadingMenu(token));
        },
        logOut: () => {
            dispatch(AuthActions.signOut());
        },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Index);
