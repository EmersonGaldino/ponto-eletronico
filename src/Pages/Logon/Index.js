import React, {useState, useEffect} from 'react';
import {useHistory, Link} from 'react-router-dom';
import {connect} from 'react-redux';
import Lottie from "react-lottie";
import {
    Button,
    Form,
    Grid,
    Header,
    Message,
    Segment,
    Icon,
} from 'semantic-ui-react';

import FormCreateUser from '../CreateUser/index'
import * as AuthOnActions from '../../store/auth/actions';
import * as MenuActions from '../../store/menu/actions';
import HomeAnimation from '../../Components/Lottie/39998-web-development.json'
import {DisplayMessage, Container, DivLottie, DivLogin} from './styles';

const Logon = ({
                   loadingMenu,
                   signInRequest,
                   loading,
                   success,
                   message
               }) => {
    const defaultOptions = {
        loop: true,
        autoplay: true,
        animationData: HomeAnimation,

    };
    const [login, setLogin] = useState(null);
    const [password, setPassword] = useState(null);
    const [load, setLoad] = useState(null);
    const [erro, setErro] = useState(false);
    const [msg, setMsg] = useState(null);

    const [visibleForm, setVisibleForm] = useState(false);

    const history = useHistory();

    async function handlerLogin() {
        let valid = null;

        if (login === '' && password === '') valid = 0;
        else if (login === '') valid = 1;
        else if (password === '') valid = 2;

        switch (valid) {
            case 0:
                setMsg(`Precisamos do seus dados para acessar.`);
                setErro(true);
                setLoad(null);
                return false;
                break;
            case 1:
                setMsg(`Login não informado.`
                );
                setErro(true);
                setLoad(null);
                return false;
                break;
            case 2:
                setMsg(
                    `Senha não informada.`
                );
                setErro(true);
                setLoad(null);
                return false;
                break;
            default:
                break;
        }
        await signInRequest({login, password})
    }

    useEffect(() => {
        if (success) {
            loadingMenu();
            history.push('/home');
        }
    }, [success])


    return (
        <Container>
            <DivLottie>
                <Lottie options={defaultOptions}/>
            </DivLottie>
            <DivLogin>
                <Grid
                    textAlign="center"
                    style={{height: '100vh'}}
                    verticalAlign="middle"
                >
                    <Grid.Column style={{maxWidth: 450}}>
                        {!visibleForm ? (
                            <>
                                <Header as="h2" color="blue" textAlign="center">
                                    <Icon name="clock outline"/> Realize o login
                                </Header>
                                <Form size="large">
                                    <Segment>
                                        <Form.Input
                                            fluid
                                            icon="user"
                                            iconPosition="left"
                                            placeholder="E-mail address"
                                            onChange={(e) => setLogin(e.target.value)}
                                        />
                                        <Form.Input
                                            fluid
                                            icon="lock"
                                            iconPosition="left"
                                            placeholder="Password"
                                            type="password"
                                            onChange={(e) => setPassword(e.target.value)}
                                        />

                                        {msg !== null || message !== '' ?
                                            <DisplayMessage>
                                                <li>
                                                    <ul>{msg !== null ? msg : message}</ul>
                                                </li>
                                            </DisplayMessage> : ''}
                                        <Button
                                            fluid
                                            inverted
                                            color='blue'
                                            onClick={() => handlerLogin()}
                                            loading={loading}>
                                            Logar
                                        </Button>
                                    </Segment>
                                    <Message>
                                        Ainda não tem acesso? <Link to='/createUser'>Crie sua conta</Link>
                                    </Message>
                                </Form>
                            </>
                        ) : (<FormCreateUser/>)}

                    </Grid.Column>

                </Grid>
            </DivLogin>
        </Container>
    );
};

const mapStateToProps = (state) => {
    return {
        token: state.auth.token,
        loading: state.auth.loading,
        success: state.auth.success,
        message: state.auth.message
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        loadingMenu: (token) => {
            dispatch(MenuActions.loadingMenu(token));
        },
        signInRequest: (params) => {
            dispatch(AuthOnActions.signInRequest(params));
        },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Logon);
