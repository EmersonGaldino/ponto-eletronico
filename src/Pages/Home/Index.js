import React, {useEffect} from 'react';

import Menu from '../../Components/Menu/Index';
import * as PointActions from "../../store/Point/actions";
import {connect} from "react-redux";
import moment from "moment";
import {isAutenticed} from '../../common/AuthHelper';
import {useHistory} from 'react-router-dom';

import News from '../News/index'

import 'react-toastify/dist/ReactToastify.css';

const Home = ({loadingPoint}) => {
    const history = useHistory();
    const monthActual = moment().format("M")
    useEffect(() => {
        let auth = isAutenticed();

        if (!auth) {
            history.push('/');
        }
        loadingPoint(monthActual)
    }, [])
    return (
        <>
            <Menu/>
            <News/>
        </>
        );
}
const mapStateToProps = (state) => {
    return {
        success: state.point.success,
        userInfo: state.auth.userInfo,
        loading: state.menu.loading,
        points: state.point.points?.itens
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        loadingPoint: (month) => {
            dispatch(PointActions.loadingPoint(month));
        },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);