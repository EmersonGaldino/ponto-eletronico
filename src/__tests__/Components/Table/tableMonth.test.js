import React from "react";
import {shallow, ReactWrapper} from 'enzyme';
import Index from '../../../Components/MonthsTable/index'
import {Provider} from 'react-redux';
import store from '../../../store/Configuration/store';

describe('Testing MonthsTable component', () => {

    it('Should valid if number is two', () => {
        expect(2).toMatchSnapshot()
    })

    it('Should render correctly', () => {
        // const wrapper = shallow(
        //     <Provider store={store}>
        //         <Index month={12}
        //                index={1}
        //                loadingPoint={12}
        //                points={[{
        //                    backlunch: "13:31",
        //                    breaklunch: "12:31",
        //                    date: "0001-01-01T00:00:00",
        //                    entrance: "01/12/2020 08:33:48",
        //                    input: "08:33",
        //                    out: "18:02",
        //                    outPut: "01/12/2020 18:02:01"
        //                }]}
        //                loading={false}
        //                success={true}
        //                totalHours={90}
        //         />
        //     </Provider>
        // )
        // expect(wrapper).toMatchSnapshot();
    })
})
