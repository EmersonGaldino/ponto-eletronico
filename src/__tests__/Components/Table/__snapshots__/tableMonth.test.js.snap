// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MonthsTable component Should valid if number is two 1`] = `2`;
